### ISSUE TITLE
[Suggestion] Show Album Cover in Rich Presence

### ISSUE THREAD

**Issue by ghost** [A-da-m/AppleMusicPresence#4]
<img width="406" alt="Screen Shot 2020-05-01 at 1 18 21 PM" src="https://user-images.githubusercontent.com/32424458/80825425-5b446200-8bae-11ea-809c-1ad012049477.png">
⬆️ my ideal user interface

**Comment by A-da-m** [comment]
I'll look into it but from what I've seen it wouldn't easily possible, and it would require that you make your own discord application for it to work.

**Comment by ghost** [comment]
I wouldn't be opposed to creating my own Discord application- it isn't super difficult and could be implemented as an optional feature if people want to include album art.

**Comment by A-da-m** [comment]
Yeah, that could work. Another issue I've run into is discord caches the images so sometimes the album art would be one song behind or even more.

**Comment by mattgeowild** [comment]
I would look into how iTunesRichPresence and similar apps have done it. 👍

**Comment by ghost** [comment]
I've done more researched into this subject, and I don't think this is actually possible at all on a large scale. As far as I can tell, the only images that can be used must be pre-uploaded to the Discord Developer Portal, in the Rich Presence Assets section. The only way I can think to do this would take quite a bit of time and effort, and would definitely require the user to create their own application -- and would still have pretty large limitations.

The Rich Presence Assets page has a max of 300 assets. So, excluding logos, etc, you could hypothetically upload around 255 of your most commonly-played album covers and match the album name or some other identification variable against an array of titles/ids and album covers. 

I'm going to give this a shot. I don't see why it wouldn't work, it's just a really sucky solution until Discord decides to add Apple Music support, if ever.

**Comment by ghost** [comment]
Using external images is now supported in the API. This is now possible @A-da-m
